woocommerce/woocommerce
# Remove all references to the product add to cart block.

### Changes proposed in this Pull Request:

Closes #44503 and ensures that https://github.com/woocommerce/woocommerce-blocks/issues/12032 is not re-introduced.

Also closes #46061 

When I dug into why https://github.com/woocommerce/woocommerce-blocks/issues/12032  happens I found that the runtime exception is when it tries to call `objectHasProp` of the `@woocommerce/types` library. It **seems** that when we removed the product add to cart block, it somehow changed webpack's bundling of the types library and Webpack mistakenly tree-shakes out `objectHasProp` alongside a bunch of the other exported functions. Product add to cart block certainly made extensive use of that package, but I can't quite explain why it's being tree-shaken.

The package is mostly type definition's with a few functions such as `objectHasProp` so I've marked it as having side effects to tell webpack not to tree-shake it and this immediately fixed the problem. 



### How to test the changes in this Pull Request:

I found I could reproduce https://github.com/woocommerce/woocommerce-blocks/issues/12032 in the same way, so you should test the steps there and see if you can reproduce.

Roughly:

1. Build the monorepo, then navigate to `plugins/woocommerce-blocks` and run `pnpm build`
2. Create a page or post with the mini cart block and the all products block together.
3. Go to your new page and open the mini cart
4. There should be no error. 



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
Remove experimental unused product add to cart block, don't tree-shake `@woocommerce/types` to avoid runtime exception.

</details>
